m2/interface/storage.py

```
@@ -0,0 +1,131 @@
+"""Storage Interface"""
+
+from abc import ABCMeta, abstractmethod
+
+
+class Storage(object):
+    """
+    Abstract class for the Storage Interface.
+    Only contains the necessary functionality of
+    the Storage drivers.
+    """
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def copy(self, image_id, copy_name):
+        """
+        Make a deep copy of a given image.
+
+        :param image_id: Id of the image to be copied
+        :param copy_name: Name of the image copy
+        :return: Id of the image copy
+        """
+        pass
+
+    @abstractmethod
+    def clone(self, instance_id, parent_image_id):
+        """
+        Make a clone of a given image.
+
+        :param instance_id: Id of the instance for which
+                            the clone is used to provision
+        :param parent_image_id: Id of the image to clone
+        :return: Id of the clone image
+        """
+        pass
+
+    @abstractmethod
+    def delete_clone(instance_id):
+        """
+        Delete a clone of a given image.
+
+        :param instance_id: Id of the instance provisioned from the clone
+        :return: None
```

based on what I learnt from sourabh: maybe raise an exception in case of failed clone.

---

and same comment about the other methods that return None.

---

I was considering that, I guess I was unsure if the types of exceptions we want to raise should be discussed among the group.